### ISSUE TITLE
Broken Python attribute protocol.

### ISSUE THREAD

**Issue by rysson** [feluxe/sty#17]
Hi.

It's a very nice module. Could you fix attribute protocol?
You have:

```python
    __getattr__: Callable = dict.__getitem__
    __setattr__: Callable = dict.__setitem__
    __delattr__: Callable = dict.__delitem__
```

and now I have:

```python
>>> from sty import ef
>>> print(getattr(ef, 'underline', ef.underl))
KeyError: 'underline'
```

`__getitem__` should raise `KeyError` and `__getattr__` should raise `AttributeError`.

I know, I can use item protocol (`ef['underline']` or `'underline' in ef`), but object should API be pythonic IMHO.

**Comment by feluxe** [comment]
That's a good point.

I think we could replace:

```python
    __getattr__: Callable = dict.__getitem__
    __setattr__: Callable = dict.__setitem__
    __delattr__: Callable = dict.__delitem__
```

with  this:

```python

    def __getattr__(self, k):
        try:
            return object.__getattribute__(self, k)
        except AttributeError:
            try:
                return self[k]
            except KeyError:
                raise AttributeError(k)

    def __setattr__(self, k, v):
        try:
            object.__getattribute__(self, k)
        except AttributeError:
            try:
                self[k] = v
            except:
                raise AttributeError(k)
        else:
            object.__setattr__(self, k, v)

    def __delattr__(self, k):
        try:
            object.__getattribute__(self, k)
        except AttributeError:
            try:
                del self[k]
            except KeyError:
                raise AttributeError(k)
        else:
            object.__delattr__(self, k)
```

I just tested the above implementation and it seems to work fine. There is also no impact on access performance.

**Comment by rysson** [comment]
Great. I suggest to remove `__getattribute__` from `__getattr__`.
`__getattr__` is called only if instance has no attribute.

It could look like:

```python
    def __getattr__(self, k):
        try:
            return self[k]
        except KeyError:
            raise AttributeError(k)
```

`__getattribute__` in `__setattr__` and `__delattr__` seems to be a nice hack :-)

**Comment by feluxe** [comment]
Thanks a lot for the update! I think the points you raise make a lot of sense. We should add this to sty, with the next release. :)

**Comment by feluxe** [comment]
Fix with 6550017d5c8da45970061c0475dc1231c32ba45f
